### ISSUE TITLE
Duplicate modules via different ways of import causing `instanceof` confusion.

### ISSUE THREAD

**Issue by aq1018** [webpack/webpack#4614]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**

`webpack2` causes [this bug](https://github.com/erikras/redux-form/issues/2269#issuecomment-290887297) in `redux-form`. I think the reason is two copies of `SubmissionError` classes are created and can be referenced separately, causing `instanceof` confusions.

**If the current behavior is a bug, please provide the steps to reproduce.**

Please see the error demonstration here:

https://github.com/aq1018/webpack2-redux-form-submission-error-weirdness

**What is the expected behavior?**

`instanceof` should return true from instance of classes imported from either import methods.

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

**Comment by sokra** [comment]
`redux-form/es/SubmissionError` and `redux-form/lib/SubmissionError` are different modules.

**Comment by aq1018** [comment]
@sokra Thanks for looking into this. Just realized that webpack loads from `es` as indicated by the `module` main field. I changed the demo code mentioned above to import from `es`, and the error is no longer there.

But I think there is a different bug in my code that is not captured in my demo code above. Somehow, in my code, I used `import { SubmissionError } from 'redux-form'` and `throw new SubmissionError(...)`, but somehow in side redux code, `submitError instanceof SubmissionError` always returns false. I was able to confirm this by adding a console.log line in my `node_modules/redux-form/lib/handleSubmit`. Note, I instrumented `lib` modules. Instrumenting `es` modules yielded no effect. This leads me to think that webpack is loading from `lib` instead of `es` in my code.

I added a [gist](https://gist.github.com/aq1018/bc3835b2e151e30085d7072efd0c6151) extracted the following from my project:

* `webpack.config.bable.js`
* `.babelrc`
* `package.json`
*  `yarn.lock`

Thank you very much for your help!

**Comment by aq1018** [comment]
@sokra I think I sort of figured out the reason. I was importing `redux-form/immutable` throughout my code, but I did `import { SubmissionError } from 'redux-form'` instead of `import { SubmissionError } from 'redux-form/immutable'`.

Soon as I updated all the reference from `redux-form` to `redux-form/immutable`, everything just worked. What confused me was that webpack version 1 worked without those changes though. ( Maybe the dedupe plugin was doing its magic? No idea... )
